src/test/java/edu/harvard/iq/dataverse/engine/command/impl/ReconcilePIDCommandTest.java

```diff
@@ -0,0 +1,354 @@
+import edu.harvard.iq.dataverse.util.SystemConfig;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.TypedQuery;
+import org.assertj.core.util.Lists;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentMatchers;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+/**
+ *
+ * @author jdarms
+ */
+public class ReconcilePIDCommandTest {
+
+    TestDataverseEngine engine;
+    DataverseRoleServiceBean drsb;
+    SystemConfig systemConfig;
+    IndexServiceBean indexServiceBean;
+    RoleAssigneeServiceBean roleAssigneeServiceBean;
+    UserNotificationServiceBean notificationService;
+    @BeforeEach
+    public void setUp() {
+        drsb= mock(DataverseRoleServiceBean.class);
+        systemConfig =mock(SystemConfig.class);
+        indexServiceBean=mock(IndexServiceBean.class);
+        roleAssigneeServiceBean=mock(RoleAssigneeServiceBean.class);
+        notificationService =mock(UserNotificationServiceBean.class);
+        engine = new TestDataverseEngine(new TestCommandContext() {
+            @Override
+            public DataverseRoleServiceBean roles() {
+                return drsb;
+            }
+
+            @Override
+            public RoleAssigneeServiceBean roleAssignees() {
+                return roleAssigneeServiceBean;
+            }
+
+            @Override
+            public EntityManager em() {
+                return new NoOpTestEntityManager();
+
+            }
+
+            @Override
+            public IndexServiceBean index() {
+                return indexServiceBean;
+            }
+
+            @Override
+            public SystemConfig systemConfig() {
+                return  systemConfig;
+            }
+
+            @Override
+            public UserNotificationServiceBean notifications() {
+                return notificationService;
+            }
+        });
+    }
+
+    @Test
+    public void testSucessfullChangeWithoutFilePIDs() throws Exception {
+        // given
+        //
+        AuthenticatedUser superUser = new AuthenticatedUser();
+        superUser.setSuperuser(true);
+        DataverseRequest request = MocksFactory.makeRequest(superUser);
+        // given a PID Provider
+        PidProvider pidProvider = mock(PidProvider.class);
+        when(pidProvider.getProtocol()).thenReturn("PROTO");
+        when(pidProvider.getAuthority()).thenReturn("PROTO");
+        when(pidProvider.getShoulder()).thenReturn("PROTO");
+        when(pidProvider.getSeparator()).thenReturn("PROTO");
+        when(pidProvider.registerWhenPublished()).thenReturn(false);
+
+
+
+        // given a dataset linked with the PIDprovider
```

Comment:
FWIW - the PidUtilTest has code to setup/register PidProviders - it might not be too hard to make those callable here as well. With the change to not use effectivePidGenerator in the registerExternalIdentifier call, you'll need to at least call PidUtil.addToProviderList and have a parseable PID to get a working test.